### ISSUE TITLE
Lon and lat ranges (max of 180 vs 360 deg) in geospatial data

### ISSUE THREAD

**Issue by atrisovic** [cf-convention/cf-conventions#435]
# TLDR

In some geospatial data longitude is given in the range of 0-360 degrees, while in the others the range is -180-180 degrees. Does the CF-Convention contain information which range is used in the data, and how is it denoted?

Does this ambiguity sometimes exist for latitude also, or it is always in the range -90-90 degrees?

# Question

Hi everyone, 

We are trying to automatically extract metadata from geospatial files (netCDF and HDF5) and populate that information in the geospatial metadata fields in the Dataverse (dataverse.org) data repository. In particular, we have a `geographicBoundingBox` metadata field that we want to populate using the info from the standard fields `geospatial_lat_min`, `geospatial_lat_max`, `geospatial_lon_min`, `geospatial_lon_max` that are typically found within these data. 

The issue is that `lon` is sometimes given in the range of 0-360, and sometimes in the range of -180-180 degrees.
Does the CF-convention denote which range is used for `lon` and `lat` in the data?

From what I've seen, if the `units` attribute is set to `degrees_east`, then the range is typically 0-360 for longitude, while if the `units` is `degrees_west` the range is -180-180 (see examples below). Is this the general rule, or there is a better way to see the range?

Also, does this ambiguity sometimes exist for latitude, or it is always in the range -90-90 degrees?

# Example

One of our test datasets contains the following metadata:

```
 <attribute name="standard_name_vocabulary" value="CF Standard Name Table v31" />
  <attribute name="geospatial_bounds_crs" value="EPSG:4326" />
  <attribute name="geospatial_lat_min" type="double" value="41.8" />
  <attribute name="geospatial_lat_max" type="double" value="49.62" />
  <attribute name="geospatial_lon_min" type="double" value="343.68" />
  <attribute name="geospatial_lon_max" type="double" value="353.78" />
  <attribute name="geospatial_lat_units" value="degrees_north" />
  <attribute name="geospatial_lon_units" value="degrees_east" />
```

Another example contains this metadata:

```
  <attribute name="Conventions" value="CF-1.6" />
  <attribute name="geospatial_lat_min" value="25.066666666666666" />
  <attribute name="geospatial_lat_max" value="49.40000000000000" />
  <attribute name="geospatial_lon_min" value="-124.7666666333333" />
  <attribute name="geospatial_lon_max" value="-67.058333300000015" />
  <attribute name="geospatial_lon_resolution" value="0.041666666666666" />
  <attribute name="geospatial_lat_resolution" value="0.041666666666666" />
  <attribute name="geospatial_lat_units" value="decimal_degrees north" />
  <attribute name="geospatial_lon_units" value="decimal_degrees east" />
  <attribute name="coordinate_system" value="EPSG:4326" />
  <attribute name="geospatial_bounds" value="POLYGON((-124.7666666333333 49.400000000000000, -124.7666666333333 25.066666666666666, -67.058333300000015 25.066666666666666, -67.058333300000015 49.400000000000000, -124.7666666333333 49.400000000000000))" />
```

**Comment by taylor13** [comment]
Coordinate values in CF files must be stored monotonically increasing or decreasing, so you could extract the first and last coordinate values stored to determine the domain.

**Comment by atrisovic** [comment]
Hi @taylor13, thanks for the prompt reply!

I'm not sure I completely understand. For instance, in my examples:

`<attribute name="geospatial_lon_min" value="343.68" />` and `<attribute name="geospatial_lon_max" value="353.78" />`

and 

`<attribute name="geospatial_lon_min" value="-124.7666666333333" />` and `<attribute name="geospatial_lon_max" value="-67.058333300000015" />`

both `min` longitudes are smaller than `max` longitudes. Can I infer from that the range is 0-360?

The issue is that sometimes only a small region or a city will be captured in the data and for example the region at the longitude of (50, 60) could work for both ranges (0-360 and -180-180).

**Comment by JonathanGregory** [comment]
Closing this issue because the question has been answered

**Comment by gerritholl** [comment]
I know this is old, but if the metadata say `<attribute name="coordinate_system" value="EPSG:4326" />`, then it must be `[-180, 180]`, because [that's the defined range for EPSG 4326](https://epsg.io/4326).
